From 493f1ea80e6d3249ec0dce073b0e57b00c831561 Mon Sep 17 00:00:00 2001
From: Hendrik Brueckner <brueckner@redhat.com>
Date: Wed, 18 Jun 2014 10:27:50 -0400
Subject: [s390] compat: fix PSW32_USER_BITS definition

Message-id: <1403087280-902-5-git-send-email-brueckner@redhat.com>
Patchwork-id: 84242
O-Subject: [RHEL7.1 PATCH 04/14] [s390] s390/compat: fix PSW32_USER_BITS definition
Bugzilla: 1099147
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Jes Sorensen <Jes.Sorensen@redhat.com>
RH-Acked-by: Steve Best <sbest@redhat.com>

Description
-----------
commit 075dfd82102d2048e43e1cbf48d558d915c50072
Author: Heiko Carstens <heiko.carstens@de.ibm.com>
Commit: Martin Schwidefsky <schwidefsky@de.ibm.com>

    s390/compat: fix PSW32_USER_BITS definition

    PSW32_USER_BITS should define the primary address space for user space
    instead of the home address space.
    Symptom of this bug is that gdb doesn't work in compat mode.

    The bug was introduced with e258d719ff28 "s390/uaccess: always run the kernel
    in home space" and f26946d7ecad "s390/compat: make psw32_user_bits a constant
    value again".

    Cc: stable@vger.kernel.org # v3.13+
    Reported-by: Andreas Arnez <arnez@linux.vnet.ibm.com>
    Signed-off-by: Heiko Carstens <heiko.carstens@de.ibm.com>
    Signed-off-by: Martin Schwidefsky <schwidefsky@de.ibm.com>

Bugzilla
--------
BZ 1099147
https://bugzilla.redhat.com/show_bug.cgi?id=1099147

Upstream status of the patch
----------------------------
The patch is upstream as of kernel version v3.14-rc1
http://git.kernel.org/?p=linux/kernel/git/torvalds/linux-2.6.git;a=commitdiff;h=075dfd82102d2048e43e1cbf48d558d915c50072

Brew
----
https://brewweb.devel.redhat.com/taskinfo?taskID=7588402

Test status
-----------
The patch has been tested and fixes the problem.
The fix has been verified by the IBM test department.

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/s390/include/asm/compat.h b/arch/s390/include/asm/compat.h
index 4bf9da0..5d7e8cf 100644
--- a/arch/s390/include/asm/compat.h
+++ b/arch/s390/include/asm/compat.h
@@ -38,7 +38,8 @@
 
 #define PSW32_USER_BITS (PSW32_MASK_DAT | PSW32_MASK_IO | PSW32_MASK_EXT | \
     PSW32_DEFAULT_KEY | PSW32_MASK_BASE | \
-    PSW32_MASK_MCHECK | PSW32_MASK_PSTATE | PSW32_ASC_HOME)
+    PSW32_MASK_MCHECK | PSW32_MASK_PSTATE | \
+    PSW32_ASC_PRIMARY)
 
 #define COMPAT_USER_HZ  100
 #define COMPAT_UTS_MACHINE "s390\0\0\0\0"
-- 
1.7.1